### ISSUE TITLE
Images in headings not slugged correctly

### ISSUE THREAD

**Issue by ajoga** [remarkjs/remark-validate-links#59]
### Subject of the issue

I found a case where a working link in github is detected as broken by remark-validate-links when the linked content has an image in it, like the badges in READMEs.

### Your environment

*   **OS**: debian 10
*   **Packages**: remark-cli remark-validate-links
*   **Env**: npm 6.10.3, remark-cli@8.0.1, remark-validate-links@10.0.2, Firefox 81.0 (64 bits)

### Steps to reproduce

```
$ git clone https://github.com/ajoga/electricitymap-contrib.git
$ git checkout -b remark-validate-links-br
$ ./node_modules/.bin/remark --frail -u validate-links .
test.md
  106:71-106:107  warning  Link to unknown heading: `electricitymap---`  missing-heading  remark-validate-links
   108:71-108:96  warning  Link to unknown heading: `hello-`             missing-heading  remark-validate-links
   110:71-110:97  warning  Link to unknown heading: `hello2-`            missing-heading  remark-validate-links

⚠ 3 warnings

```

### Expected behavior

These three warning should not appear. You can browse here to see that the three links are working : https://github.com/ajoga/electricitymap-contrib/blob/remark-validate-links-br/test.md

### Actual behavior

We get warnings.

### Backstory / use-case

I'm using remark-validate-links on the README of https://github.com/tmrowco/electricitymap-contrib 

I've seen that often badges are on the line below the h1, but I guess having them in the h1 is not uncommon either, hence the report

**Comment by wooorm** [comment]
Heya, nice catch! The problem is the slugging of images in headings. It’s most likely also present in remark-slug, mdast-util-toc!

**Comment by ajoga** [comment]
Thank you for the triage :)

**Comment by wooorm** [comment]
Btw: to link to the top, you can also use `#readme`. That works in all readmes on GH. And I think it’s supported here too

**Comment by ajoga** [comment]
> Btw: to link to the top, you can also use `#readme`. That works in all readmes on GH. And I think it’s supported here too

TIL, thanks :-) 

Reference for future googlers : 

> GitHub's readme div has id="readme" and this makes browsers scroll down to where the readme starts.

 https://gitlab.com/gitlab-org/gitlab-foss/-/issues/34945

**Comment by wooorm** [comment]
Well, this wasn’t as easy as I thought. I reverse engineered how GitHub slugs, again, today: https://github.com/Flet/github-slugger/pull/38. Hopefully can pull this through to this project soonish!
